src/cli.rs

```diff
@@ -7,6 +7,29 @@ use niri_ipc::{Action, OutputAction};
 
 use crate::utils::version;
 
+#[derive(Clone, Debug, clap::ValueEnum)]
+pub enum CompletionShell {
+    Bash,
+    Elvish,
+    Fish,
+    PowerShell,
+    Zsh,
+    Nushell,
+}
+
+impl From<CompletionShell> for Shell {
+    fn from(shell: CompletionShell) -> Self {
+        match shell {
+            CompletionShell::Bash => Shell::Bash,
+            CompletionShell::Elvish => Shell::Elvish,
+            CompletionShell::Fish => Shell::Fish,
+            CompletionShell::PowerShell => Shell::PowerShell,
+            CompletionShell::Zsh => Shell::Zsh,
+            CompletionShell::Nushell => panic!("Nushell should be handled separately"),
+        }
```

Comment:
Is [the relevant trait from clap_complete](https://docs.rs/clap_complete/latest/clap_complete/aot/trait.Generator.html) dyn-compatible? If so, I think it would be cleaner to remove this From, instead inline the match into the code below, returning a `&dyn Shell::Bash` and so on, as well as `&dyn Nushell`, which would all fit into a single variable of type `&dyn Generator`. Then there would be only one call to `clap_complete::generate()`.